src/WebJobs.Script.WebHost/Management/LinuxSpecialization/RunFromPackageHandler.cs

```diff
@@ -51,8 +53,31 @@ public async Task<bool> ApplyBlobPackageContext(RunFromPackageContext pkgContext
                         EnvironmentSettingNames.DefaultLocalSitePackagesPath)
                     : string.Empty;
 
-                // download zip and extract
-                var filePath = await _packageDownloadHandler.Download(pkgContext);
+                string filePath;
+
+                if (!pkgContext.IsRunFromLocalPackage())
+                {
+                    // download zip
+                    filePath = await _packageDownloadHandler.Download(pkgContext);
+                }
+                else
+                {
+                    _logger.LogInformation($"{nameof(ApplyRunFromPackageContext)}: Going to copy package file.");
+                    if (!azureFilesMounted)
+                    {
+                        _logger.LogWarning($"{nameof(ApplyRunFromPackageContext)} failed. FileShareMount is required when RunFromPackage is 1.");
+                    }
+
+                    // copy zip
+                    filePath = _packageCopyHandler.CopyPackageFile();
```

Comment:
>CopyPackageFile [](http://example.com/codeflow?start=51&length=15)

can this be a private method in PackageDownloadHandler that will be used by _packageDownloadHandler.Download if required?

Reply:
Yes, I moved the method to `PackageDownloadHandler `.